frontend/src/store/engagementSlice.ts

```diff
@@ -11,15 +11,19 @@ import { get } from '../services/apiService';
 import { toDate } from '../utils/date';
 import { environment } from '../utils/environment';
 import { AppThunk, RootState } from './reducer';
+import { getRole } from '../services/authService';
+import { isManagerRole } from '../utils/role';
 
 const { baseUrl } = environment;
 
 const engagementAdapter = createEntityAdapter<Engagement>({
   selectId: ({ staffId, start }) => `${staffId}::${start}`,
 });
 
-const engagementsFetchUrl = (start: Date, end: Date) =>
-  `${baseUrl}/staff/shifts?start=${start.toISOString()}&end=${end.toISOString()}`;
+const engagementsFetchUrl = async (start: Date, end: Date, token?: string) =>
+  isManagerRole(await getRole(token))
+    ? `${baseUrl}/staff/shifts?start=${start.toISOString()}&end=${end.toISOString()}`
+    : `${baseUrl}/myShifts?start=${start.toISOString()}&end=${end.toISOString()}`;
```

Comment:
For me it could be the same endpoint. But its not something that needs changed now.